sdk/python/packages/flet-core/src/flet_core/page.py

```diff
@@ -916,6 +914,8 @@ def __on_page_change_event(self, data: str) -> None:
                 for name in props:
                     if name != "i":
                         self._index[id]._set_attr(name, props[name], dirty=False)
+                        if id in self.__snapshot:
+                            self.snapshot[id][name] = props[name]
```

Comment:
I've fixed that here already: https://github.com/flet-dev/flet/pull/3854
You may leave only `form_field.dart` fix in your PR.